### PR TITLE
Fix `test_published_images.yml` [DI-316]

### PR DESCRIPTION
**Pull request by JackPGreen**
The action added in https://github.com/hazelcast/hazelcast-docker/pull/814 could not be properly tested via GitHub until merged and at that point additional defects were discovered.

[Example of now successful completion](https://github.com/hazelcast/hazelcast-docker/actions/runs/11943342729).

Post-merge actions:
- [ ] backport this + previous PRs to maintenance branches
- [ ] update [release documentation to reference](https://hazelcast.atlassian.net/wiki/spaces/EN/pages/4681302104/Create+release+5.x#Docker-release-tag-test)

Fixes: [DI-316](https://hazelcast.atlassian.net/browse/DI-316)

[DI-316]: https://hazelcast.atlassian.net/browse/DI-316?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/test_published_images.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         variant: 
           - ''
           - 'slim'
-        distribution_type: ${{ fromJson(needs.set-matrix.outputs.distribution-type-matrix) }}
+        distribution-type: ${{ fromJson(needs.set-matrix.outputs.distribution-type-matrix) }}
```
Comment (ldziedziul) [inline .github/workflows/test_published_images.yml:71]: why this rename?

**Comment by JackPGreen** [comment]
> > The action added in #814 could not be properly tested via GitHub until merged
> 
> You could simply merge an empty (or minimalistic) workflow to the master, and then test the workflow on the branch

I wasn’t sure what the eventual inputs would be (which is good as we changed them right at the end) and was worried GitHub would only let me use the inputs on `master`.

File: .github/workflows/test_published_images.yml (hunk 1 above)
Comment (JackPGreen) [reply .github/workflows/test_published_images.yml:71]: > why this rename?

Consistency between the matrix name and its source. Probably not required but neater. Can revert if problematic…

**Review by nishaatr — APPROVED** [review]

**Comment by ldziedziul** [comment]
> I wasn’t sure what the eventual inputs would be (which is good as we changed them right at the end) and was worried GitHub would only let me use the inputs on `master`.

I'm pretty sure it will show input from the PR branch

**Review by ldziedziul — APPROVED** [review]

**Pull request merged**